homeassistant/components/teslemetry/button.py

```diff
@@ -0,0 +1,114 @@
+"""Button platform for Teslemetry integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from itertools import chain
+
+from tesla_fleet_api.const import Scope
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import TeslemetryVehicleEntity
+from .models import TeslemetryVehicleData
+
+
+@dataclass(frozen=True, kw_only=True)
+class TeslemetryButtonEntityDescription(ButtonEntityDescription):
+    """Describes a Teslemetry Button entity."""
+
+    func: Callable | None = None
```

Comment:
Can we type the Callable?

Reply:
Changed to Callable[[TeslemetryButtonEntity], Any]

Reply:
Done